main.tf

```diff
@@ -16,19 +16,22 @@
 
 locals {
   tmp_credentials_path = "${path.module}/terraform-google-credentials.json"
-  cache_path           = "${path.module}/cache/${random_id.cache.hex}"
+  cache_path           = local.skip_download ? "" : "${path.module}/cache/${random_id.cache[0].hex}"
   gcloud_tar_path      = "${local.cache_path}/google-cloud-sdk.tar.gz"
   gcloud_bin_path      = "${local.cache_path}/google-cloud-sdk/bin"
   gcloud_bin_abs_path  = abspath(local.gcloud_bin_path)
   components           = join(",", var.additional_components)
 
-  gcloud              = var.skip_download ? "gcloud" : "${local.gcloud_bin_path}/gcloud"
+  download_override = data.external.env_override.result.download
```

Comment:
As the `data.external.env_override.result.download` is just used for reading if GCLOUD_TF_DOWNLOAD is set does it make sense to just make it a TF variable? This way we can natively set it with `TF_VAR_*`.

Reply:
TF variables are only applied at the root module/workspace. We want to be able to use a global variable without having to pass it through modules.

Reply:
Ah yes that escaped me :)